src/measure_theory/function/lp_space.lean

```diff
@@ -754,6 +773,56 @@ begin
   exact snorm'_add_lt_top_of_le_one hf.1 hg.1 hf.2 hg.2 hp_pos hp1_real,
 end
 
+section map_measure
+
+variables {β : Type*} {mβ : measurable_space β} {f : α → β} {g : β → E}
+
+include mβ
+
+lemma snorm_ess_sup_map_measure (hg : ae_measurable g (measure.map f μ)) (hf : measurable f) :
+  snorm_ess_sup g (measure.map f μ) = snorm_ess_sup (g ∘ f) μ :=
+ess_sup_map_measure hg.ennnorm hf
+
+lemma snorm_map_measure (hg : ae_measurable g (measure.map f μ)) (hf : measurable f) :
+  snorm g p (measure.map f μ) = snorm (g ∘ f) p μ :=
+begin
+  by_cases hp_zero : p = 0,
+  { simp only [hp_zero, snorm_exponent_zero], },
+  by_cases hp_top : p = ∞,
+  { simp_rw [hp_top, snorm_exponent_top],
+    exact snorm_ess_sup_map_measure hg hf, },
+  simp_rw snorm_eq_lintegral_rpow_nnnorm hp_zero hp_top,
+  rw lintegral_map' (hg.ennnorm.pow_const p.to_real) hf,
+end
+
+lemma mem_ℒp_map_measure_iff (hg : ae_measurable g (measure.map f μ)) (hf : measurable f) :
+  mem_ℒp g p (measure.map f μ) ↔ mem_ℒp (g ∘ f) p μ :=
+by simp [mem_ℒp, snorm_map_measure hg hf, hg.comp_measurable hf, hg]
+
+lemma _root_.measurable_embedding.snorm_map_measure_of_ne_top {g : β → F}
+  (hf : measurable_embedding f) (hp : p ≠ ∞) :
```

Comment:
Do you really need to exclude `p = ∞` here? I have the impression that with `snorm_map_measure` and the fact that measurable embeddings respect ae measurability, you have all the tools to do it in all cases, no?

Reply:
That might well be true. I cannot use `snorm_map_measure` here because `F` does not have a `measurable_space`, but there could be a version with `E` (which is a measurable space) instead of `F` as in `snorm_map_measure`, for which I can remove the hypothesis about `p`. Or there could be a version that works for `F`. Thanks for pointing that out, I'll look at it.

Reply:
I removed the assumption `hp : p ≠ ∞`.